builtin/k8s/platform.go

```diff
@@ -125,6 +128,164 @@ func (p *Platform) getClientset() (*clientsetInfo, error) {
+		deploymentResource.Health = deployHealth
+		deploymentResource.HealthMessage = fmt.Sprintf("%s: %s", mostRecentCondition.Type, mostRecentCondition.Message)
+		deploymentResource.StateJson = string(deployStateJson)
+	}
+
+	// Get pod status
+
+	podClient := clientset.Clientset.CoreV1().Pods(namespace)
+	podLabelId := fmt.Sprintf("app=%s", deploymentState.Name)
+	podList, err := podClient.List(ctx, metav1.ListOptions{LabelSelector: podLabelId})
+	if err != nil {
+		return status.Errorf(codes.FailedPrecondition, "error listing pods to determine application health: %s", err)
+	}
+	if podList == nil {
+		return status.Errorf(codes.FailedPrecondition, "kubernetes pod list cannot be nil")
+	}
+
+	for _, pod := range podList.Items {
+		// Redact env vars because they can contain secrets
+		for i := 0; i < len(pod.Spec.Containers); i++ {
+			pod.Spec.Containers[i].Env = []corev1.EnvVar{}
+		}
+
+		podJson, err := json.Marshal(map[string]interface{}{
+			"pod":       pod,
+			"hostIP":    pod.Status.HostIP,
+			"ipAddress": pod.Status.PodIP,
+		})
+		if err != nil {
+			return status.Errorf(codes.Internal, "failed to marshal k8s pod definition to json: %s", podJson)
+		}
+
+		var health sdk.StatusReport_Health
+		var healthMessage string
+
+	OUTER:
+		switch pod.Status.Phase {
+		case corev1.PodPending:
+			health = sdk.StatusReport_ALIVE
+		case corev1.PodRunning:
+			// Extra checks on the latest condition to ensure pod is reporting ready and running
+			for _, c := range pod.Status.Conditions {
+				if c.Status == corev1.ConditionTrue && c.Type == corev1.PodReady {
+					health = sdk.StatusReport_READY
+					healthMessage = fmt.Sprintf("ready")
+					break OUTER
+				}
+			}
+			health = sdk.StatusReport_ALIVE
+		case corev1.PodSucceeded:
+			// kind of a weird one - in our current model pods are always assumed to be long-lived. If a pod exits at all, it's Down.
+			health = sdk.StatusReport_DOWN
+		case corev1.PodFailed:
+			health = sdk.StatusReport_DOWN
+		case corev1.PodUnknown:
+			health = sdk.StatusReport_UNKNOWN
+		default:
+			health = sdk.StatusReport_UNKNOWN
+		}
+
+		// If we don't have anything better, the pod status phase is an OK health message
+		// NOTE(izaak): An alternative health message could be the "type" of all conditions tied for the most recent
+		// latestTransitionTime concatenated together.
+		if healthMessage == "" {
+			healthMessage = fmt.Sprintf("%s", pod.Status.Phase)
+		}
+
+		sr.Resources = append(sr.Resources, &sdk.StatusReport_Resource{
+			Name:                pod.ObjectMeta.Name,
+			Id:                  fmt.Sprintf("%s", pod.UID),
+			Type:                "pod",
+			ParentResourceId:    deploymentResource.Id,
+			Health:              health,
+			HealthMessage:       healthMessage,
+			Platform:            platformName,
+			CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE,
+			StateJson:           string(podJson),
+			CreatedTime:         timestamppb.New(pod.CreationTimestamp.Time),
+		})
+	}
+
```

Comment:
We should close off the UX loop here and update the stepgroup to say we've finished building the report! Maybe something like

> Finished building report for kubernetes deployment resource